<a href="https://colab.research.google.com/github/nsk31512/terver/blob/hometask1/%D0%A2%D0%B5%D1%80%D0%B2%D0%B5%D1%80_%D0%94%D0%B71.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Задача 1

Имеется колода в 52 карты. Найти число возможностей вытянуть из неё 4 карты так, чтобы среди них был хотя бы один туз.

In [1]:
import numpy as np

Для решения задачи воспользуемся операцией сочетания $С_n^k=\frac{A_n^k}{P_n^k}=\frac{n!}{k(n-k)!}$

In [5]:
def combination(n, k):
  A_n_k = np.math.factorial(n)
  P_k = np.math.factorial(k)*np.math.factorial(n-k)
  return A_n_k//P_k  

In [20]:
result = 0
for i in range(1, 5):
  result += combination(4, i)*combination(48, 4-i)
result

76145

Ответ: 76145 возможностей вытянуть 4 карты так, чтобы среди них был хотя бы один туз.

# Задача 2 и 3

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Случайным образом студент вытягивает 3 вопроса. Какова вероятность, что все выбранные вопросы знакомы студенту? Какова вероятность что два из трёх вопросов знакомы студенту?

Для решения задачи найдем сочетания, при котором мы вытянули 3 билета из знакомых 50 и 3 билета из всех 60.

In [19]:
C_3_60 = combination(60, 3)
C_3_50 = combination(50, 3)
result_1 = C_3_50/C_3_60
result_1

0.5727644652250146

Ответ: Вероятность вытянуть 3 знакомых вопроса: 0.5727644652250146

Для второго случая нужно вычислить комбинацию событий, при котором наступит вероятность вытинуть 2 билета из 50 выученных, и 1 билет из оставшихся 10.

In [22]:
C_2_50 = combination(50, 2)*combination(10, 1)
result2 = C_2_50/C_3_60
result2

0.35797779076563413

Ответ: Вероятность вытянуть 2 знакомых вопроса из 3: 0.35797779076563413

#Задача 4

Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.

Вопрос: какова вероятность, что вы действительно больны ей?

In [25]:
#Вероятность иметь болезнь
p_H1 = 0.001

#Вероятность обратного события
p_H2 = 1-p_H1
p_H2

0.999

In [27]:
#Вероятность положительного теста
p_A_H1 = 0.99

#Вероятность ложноположительного результата
p_A_H2 = 0.01

In [28]:
#Вероятность наступления положительного события
p_A = p_A_H1*p_H1+p_A_H2*p_H2
p_A

0.01098

In [29]:
#Вероятность иметь болезнь
p_H1_A = p_A_H1*p_H1/p_A
p_H1_A

0.09016393442622951